pennylane/gradients/general_shift_rules.py

```diff
@@ -126,4 +166,152 @@ def get_shift_rule(frequencies, shifts=None):
 
     coeffs = np.concatenate((coeffs, -coeffs))
     shifts = np.concatenate((shifts, -shifts))
-    return ([[coeffs[mu], 1, shifts[mu]] for mu in range(0, 2 * n_freqs)],)
+    return np.stack([coeffs, shifts])
+
+
+@functools.lru_cache()
+def generate_shift_rule(frequencies, shifts=None, order=1):
+    r"""Computes the parameter shift rule for a unitary based on its generator's eigenvalue frequency
+    spectrum.
+
+    To compute gradients of circuit parameters in variational quantum algorithms, expressions for
+    cost function first derivatives with respect to the variational parameters can be cast into
+    linear combinations of expectation values at shifted parameter values. These "gradient recipes"

```

Comment:
Maybe: the data for these gradient recipes?

Reply:
Have removed this phrase :)